internal/xds/server/runner/runner.go

```diff
@@ -112,10 +115,8 @@ func registerServer(srv controlplane_server_v3.Server, g *grpc.Server) {
 
 func (r *Runner) subscribeAndTranslate(ctx context.Context) {
 	// Subscribe to resources
-	for snapshot := range r.Xds.Subscribe(ctx) {
-		r.Logger.Info("received a notification")
-		// Load all resources required for translation
-		for _, update := range snapshot.Updates {
+	watchutil.HandleSubscription(r.Xds.Subscribe(ctx),
+		func(update watchable.Update[string, *xdstypes.ResourceVersionTable]) {
```

Comment:
any way to move the watchable defs into the callee lib ?

Reply:
Unfortunately,
```
internal/message/watchutil.go:7:6: generic type cannot be alias
```
however, since `Update` has no methods it isn't so important that it be an alias, it can just be a new child type, so I've now done that.